staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -264,6 +267,21 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 		return nil
 	})
 
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.StorageVersionAPI) {
+		// Spawn a goroutine in aggregator apiserver to update storage version for
+		// all built-in resources
+		s.GenericAPIServer.AddPostStartHookOrDie("built-in-resources-storage-version-updater", func(context genericapiserver.PostStartHookContext) error {
+			// All apiservers (aggregator-apiserver, kube-apiserver, apiextensions-apiserver)
+			// share the same generic apiserver config. The same StorageVersion manager is used
+			// to register all built-in resources when the generic apiservers install APIs.
+			s.GenericAPIServer.StorageVersion.UpdateStorageVersions(context.LoopbackClientConfig, s.GenericAPIServer.APIServerID)
+			wait.PollImmediateInfinite(10*time.Minute, func() (bool, error) {
```

Comment:
I like this to help prevent future restarts from producing "weird" effects.

Reply:
Could you elaborate more?

Reply:
Discussed offline. Updated the poststart hook to reconcile StorageVersion objects every 10 minutes.